flowapi/app/jwt_auth_callbacks.py

```diff
@@ -0,0 +1,150 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import uuid
+from flask_jwt_extended import JWTManager, get_jwt_identity
+from flask_jwt_extended.default_callbacks import (
+    default_expired_token_callback,
+    default_verify_claims_failed_callback,
+    default_invalid_token_callback,
+    default_revoked_token_callback,
+    default_unauthorized_callback,
+    default_user_identity_callback,
+)
+from quart import current_app, request, Response
+
+
+def register_logging_callbacks(jwt: JWTManager):
+    """
+    Register callbacks which emit logs to the app's access log for:
+    - expired tokens
+    - tampered tokens
+    - invalid tokens
+    - revoked tokens
+    - blacklisted tokens
+    - unauthorised access
+
+    Parameters
+    ----------
+    jwt : JWTManager
+        JWT manager to register callbacks for
+
+    Returns
+    -------
+    JWTManager
+        The JWT manager wth the registered callbacks
+
+    """
+
+    jwt.expired_token_loader(expired_token_callback)
+    jwt.claims_verification_failed_loader(claims_verification_failed_callback)
+    jwt.invalid_token_loader(invalid_token_callback)
+    jwt.revoked_token_loader(revoked_token_callback)
+    jwt.unauthorized_loader(unauthorized_callback)
+    return jwt
+
+
+async def expired_token_callback() -> Response:
+    """
+    Log that an access attempt was made with an expired token.
+
+    Returns
+    -------
+    Response
+    """
+
+    current_app.access_logger.error(
+        "EXPIRED_TOKEN",
+        route=request.path,
+        request_id=request.request_id,
+        user=get_jwt_identity(),
```

Comment:
Is there a reason this is `get_jwt_identity()` but the other callbacks all use `str(get_jwt_identity())`?